src/test/java/com/vinylteam/vinyl/dao/jdbc/JdbcUserPostDaoITest.java

```diff
@@ -0,0 +1,41 @@
+package com.vinylteam.vinyl.dao.jdbc;
+
+import com.vinylteam.vinyl.dao.UserPostDao;
+import com.vinylteam.vinyl.entity.UserPost;
+import com.vinylteam.vinyl.util.DatabasePreparerForITests;
+import org.junit.jupiter.api.*;
+
+import java.sql.SQLException;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class JdbcUserPostDaoITest {
+    private final DatabasePreparerForITests databasePreparer = new DatabasePreparerForITests();
+    private final UserPostDao userPostDao = new JdbcUserPostDao(databasePreparer.getDataSource());
+
+
+    @BeforeAll
+    void beforeAll() throws SQLException {
+        databasePreparer.truncateAllVinylTables();
```

Comment:
This method truncates tables shops, unique_vinyls, and offers. I see how mistake could happen - vinyl tables as all tables in out vinyl application. But that doesn't apply - just tables connected to vinyl data are truncated(vinyl_browsing_history will be added later).

Solution would be either to truncate user_posts with try{} in beforeAll/Each, or to add a truncateUserPosts() to databasePreparer.